Zend/zend_vm_def.h

```diff
@@ -5945,6 +5945,14 @@ ZEND_VM_HANDLER(181, ZEND_FETCH_CLASS_CONSTANT, VAR|CONST|UNUSED|CLASS_FETCH, CO
 				HANDLE_EXCEPTION();
 			}
 
+			if (UNEXPECTED(!(c->ce->ce_flags & ZEND_ACC_CONSTANTS_UPDATED))) {
```

Comment:
I don't think there's a need to update all class constants, you should add the necessary code below (zval_update_constant_ex)

Reply:
Very nice observation! My latest commit fixes this as well